llm-vertex.el

```diff
@@ -320,23 +320,23 @@ If STREAMING is non-nil, use the URL for the streaming API."
      :headers `(("Authorization" . ,(format "Bearer %s" (llm-vertex-key provider))))
      :data (llm-vertex--chat-request prompt)
      :on-element (lambda (element)
-                   (when-let ((response (llm-vertex--get-chat-response element)))
-                     (if (stringp response)
-                         (when (> (length response) 0)
-                           (setq streamed-text (concat streamed-text response))
-                           (llm-request-callback-in-buffer buf partial-callback streamed-text))
-                       (setq function-call response))))
+                   (if (alist-get 'error element)
+                       (llm-request-callback-in-buffer buf error-callback 'error
+                                                       (llm-vertex--error-message element))
+                     (when-let ((response (llm-vertex--get-chat-response element)))
+                       (if (stringp response)
+                           (when (> (length response) 0)
+                             (setq streamed-text (concat streamed-text response))
+                             (llm-request-callback-in-buffer buf partial-callback streamed-text))
+                         (setq function-call response)))))
      :on-success (lambda (data)
                    (llm-request-callback-in-buffer
                     buf response-callback
                     (llm-vertex--process-and-return
                      provider prompt (or function-call
                                          (if (> (length streamed-text) 0)
                                              streamed-text
-                                           (llm-vertex--get-chat-response data))))))
-     :on-error (lambda (_ data)
```

Comment:
Why remove on-error here?  My understanding is that we still need it when the connection errors out with a HTTP response code, etc.

Reply:
Yes, true. What should we do here then? Throw a user-error with the arguments code and data some included, and let the error then handled by the machinery you described above?

Reply:
No, we pass this to the user by calling the callback.  I don't think this has much to do with what I described above, which is for errors that happen during streaming.  Although, actually, I suppose we should call `error-callback`.  So I think `plz-media-type-request` should throw an error, but we need to catch it in the `llm-request-plz-async` call and make sure the error callback is called.